modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -327,39 +384,241 @@ public function insertOrderToDb($id_order, $address, $value, $bits) {
 		return true;
 	}
 
+    /**
+     * Decrypts a string using the application secret.
+     */
+    public function decrypt(string $input): string
+    {
+    	$encryption_algorithm = 'AES-256-CBC';
+    	$hashing_algorith = 'sha256';
+    	$secret = $this->getCallbackSecret();
+        // prevent decrypt failing when $input is not hex or has odd length
+        if (strlen($input) % 2 || ! ctype_xdigit($input)) {
+            return '';
+        }
+
+        // we'll need the binary cipher
+        $binaryInput = hex2bin($input);
+        $iv = substr($binaryInput, 0, 16);
+        $hash = substr($binaryInput, 16, 32);
+        $cipherText = substr($binaryInput, 48);
+        $key = hash($hashing_algorith, $secret, true);
+
+        // if the HMAC hash doesn't match the hash string, something has gone wrong
+        if (hash_hmac($hashing_algorith, $cipherText, $key, true) !== $hash) {
+            return '';
+        }
+
+        return openssl_decrypt(
+            $cipherText,
+            $encryption_algorithm,
+            $key,
+            OPENSSL_RAW_DATA,
+            $iv
+        );
+    }
+
+    /**
+     * Encrypts a string using the application secret. This returns a hex representation of the binary cipher text
+     */
+    public function encrypt(string $input): string
+    {
+		$encryption_algorithm = 'AES-256-CBC';
+		$hashing_algorith = 'sha256';
+    	$secret = $this->getCallbackSecret();
+        $key = hash($hashing_algorith, $secret, true);
+        $iv = random_bytes(16);
+
+        $cipherText = openssl_encrypt(
+            $input,
+            $encryption_algorithm,
+            $key,
+            OPENSSL_RAW_DATA,
+            $iv
+        );
+        $hash = hash_hmac($hashing_algorith, $cipherText, $key, true);
+
+        return bin2hex($iv . $hash . $cipherText);
+    }
+
```

Comment:
The order_id is becoming very large and such security is not required here.
- Let us use aes-128-cbc
- iv,  use same length as callback secret truncate/padded to 16 bytes. No need to pass it
- No need to to hmac and send it. If even attacker puts in random string it is very less likely it will decrypt to an actual order_id

This is make the encrypt order_id to be 16bytes that is enough

Reply:
Changed in 7bcabc6